lib/improver/tests/test_ensemble_copula_coupling_EnsembleCopulaCouplingUtilities.py

```diff
@@ -41,10 +41,161 @@
 import numpy as np
 
 from improver.ensemble_copula_coupling.ensemble_copula_coupling_utilities \
-    import create_percentiles, create_cube_with_percentiles
+    import (create_percentiles, create_cube_with_percentiles,
+            insert_lower_and_upper_endpoint_to_1d_array,
+            concatenate_2d_array_with_2d_array_endpoints,
+            get_bounds_of_distribution,
+            reshape_array_to_have_probabilistic_dimension_at_the_front)
+from improver.ensemble_copula_coupling.ensemble_copula_coupling_constants \
+    import bounds_for_ecdf
 from improver.tests.helper_functions_ensemble_calibration import (
     set_up_temperature_cube, set_up_spot_temperature_cube,
-    _add_forecast_reference_time_and_forecast_period)
+    _add_forecast_reference_time_and_forecast_period,
+    set_up_probability_above_threshold_temperature_cube)
+
+
+class Test_reshape_array_to_have_probabilistic_dimension_at_the_front(
+        IrisTest):
+
+    """Test the insert_lower_and_upper_endpoint_to_1d_array."""
+
+    def setUp(self):
+        """Set up temperature cube."""
+        cube = (
+            _add_forecast_reference_time_and_forecast_period(
+                set_up_temperature_cube()))
+        percentile_points = np.arange(len(cube.coord("realization").points))
+        cube.coord("realization").points = percentile_points
+        cube.coord("realization").rename("percentile")
+        self.current_temperature_forecast_cube = cube
+
+    def test_basic(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        cube = self.current_temperature_forecast_cube
+        input_array = cube.data
+        plen = len(cube.coord("percentile").points)
+        reshaped_array = (
+            reshape_array_to_have_probabilistic_dimension_at_the_front(
+                cube.data, cube, "percentile", plen))
+        self.assertIsInstance(reshaped_array, np.ndarray)
+
+    def test_size_of_array(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        cube = self.current_temperature_forecast_cube
+        input_array = cube.data
+        plen = len(cube.coord("percentile").points)
+        reshaped_array = (
+            reshape_array_to_have_probabilistic_dimension_at_the_front(
+                cube.data, cube, "percentile", plen))
+        self.assertEqual(reshaped_array.shape[0], plen)
+        self.assertEqual(reshaped_array.shape, (3, 1, 3, 3))
+
+    def test_percentile_is_not_a_dimension_coordinate(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        cube = self.current_temperature_forecast_cube
+        for cube_slice in cube.slices_over("percentile"):
+            break
+        input_array = cube_slice.data
+        plen = len(cube_slice.coord("percentile").points)
+        reshaped_array = (
```

Comment:
This doesn't have any assert statements. This should probably check the returned shape and data.